From 4e258944d1aa6230c70770e89fa6e05893d788ac Mon Sep 17 00:00:00 2001
From: Yinjun Chen <a7301c@motorola.com>
Date: Tue, 15 Sep 2015 22:01:33 -0500
Subject: [PATCH 354/959] IKQCOMSD1-2959: esdfs: ->readdir op now ->iterate

Details please refer to:
585f6e1 [readdir] introduce iterate_dir() and dir_context

Change-Id: I53e8f9849288aa157f50f04a9c621b2f297bc487
Signed-off-by: Yinjun Chen <a7301c@motorola.com>
---
 fs/esdfs/file.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/fs/esdfs/file.c b/fs/esdfs/file.c
index d727bf6f304..18038e1bfb6 100644
--- a/fs/esdfs/file.c
+++ b/fs/esdfs/file.c
@@ -59,7 +59,7 @@ static ssize_t esdfs_write(struct file *file, const char __user *buf,
 	return err;
 }
 
-static int esdfs_readdir(struct file *file, void *dirent, filldir_t filldir)
+static int esdfs_readdir(struct file *file, struct dir_context *ctx)
 {
 	int err = 0;
 	struct file *lower_file = NULL;
@@ -70,7 +70,7 @@ static int esdfs_readdir(struct file *file, void *dirent, filldir_t filldir)
 		return -ENOMEM;
 
 	lower_file = esdfs_lower_file(file);
-	err = vfs_readdir(lower_file, filldir, dirent);
+	err = iterate_dir(lower_file, ctx);
 	file->f_pos = lower_file->f_pos;
 	if (err >= 0)		/* copy the atime */
 		fsstack_copy_attr_atime(dentry->d_inode,
@@ -390,7 +390,7 @@ const struct file_operations esdfs_main_fops = {
 const struct file_operations esdfs_dir_fops = {
 	.llseek		= esdfs_file_llseek,
 	.read		= generic_read_dir,
-	.readdir	= esdfs_readdir,
+	.iterate	= esdfs_readdir,
 	.unlocked_ioctl	= esdfs_unlocked_ioctl,
 #ifdef CONFIG_COMPAT
 	.compat_ioctl	= esdfs_compat_ioctl,
-- 
2.11.0

